### PR TITLE
ARROW-8409: [R] Add R wrappers for getting and setting global CPU thread pool capacity

### PR DESCRIPTION
**Pull request by wesm**
I needed this while running some benchmarks.

### CODE REVIEW

File: r/man/enums.Rd [hunk 1]
```diff
@@ -13,25 +13,7 @@
...
-An object of class \code{FileMode} (inherits from \code{arrow-enum}) of length 3.
-
-An object of class \code{Message::Type} (inherits from \code{arrow-enum}) of length 5.
-
-An object of class \code{Compression::type} (inherits from \code{arrow-enum}) of length 9.
-
-An object of class \code{FileType} (inherits from \code{arrow-enum}) of length 4.
-
-An object of class \code{ParquetVersionType} (inherits from \code{arrow-enum}) of length 2.
-}
```
Comment (wesm) [inline r/man/enums.Rd:34]: @nealrichardson do you know what this is about?

File: r/R/config.R [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#' Return number of threads used by the global CPU thread pool in libarrow
+#'
+#' @export
```
Comment (nealrichardson) [inline r/R/config.R:20]: If you want these functions to appear on the doc website table of contents, add `cpu_count` to the list in `r/_pkgdown.yml`, probably somewhere around L130. If you don't want to publicize these functions, add a line with `#' @keywords internal` around here (the man pages will still show up on the website, just not included on the ToC). Either way is fine with me.

**Review by nealrichardson — CHANGES_REQUESTED** [review]
Seems ok, just a few suggestions on the details.

File: r/man/enums.Rd (hunk 1 above)
Comment (wesm) [reply r/man/enums.Rd:34]: That did it

File: r/R/config.R [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+cpu_count <- function() {
```
[the remark this answers is not in the document]
Comment (wesm) [reply r/R/config.R:21]: Indeed, they are just the raw Rcpp functions. I had written tests for these but apparently forgot to check it in. Adapted

File: r/R/config.R [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+#' Manage the global CPU thread pool in libarrow
+#'
+#' @export
+cpu_count <- function() {
+  GetCpuThreadPoolCapacity()
+}
+
+#' @rdname cpu_count
+#'
```
Comment (nealrichardson) [inline r/R/config.R:26]:
```suggestion
```

File: r/_pkgdown.yml [hunk 5]
```diff
@@ -132,6 +132,10 @@ reference:
   - FileInfo
   - FileStats
   - FileSelector
+- title: Configuration
+  contents:
+  - cpu_count
+  - set_cpu_count
```
Comment (nealrichardson) [inline r/_pkgdown.yml:138]:
```suggestion
```

They're documented in the same file so you only include them once.

**Review by nealrichardson — APPROVED** [review]
LGTM, thanks @wesm

**Pull request closed without merging**